src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -98,19 +108,62 @@ private String getHIUId(ConsentDetail consentDetail) {
 
     public Mono<ConsentApprovalResponse> approveConsent(String authorization,
                                                         String requestId,
-                                                        List<GrantedConsent> grantedConsents) {
+                                                        List<GrantedConsent> grantedConsents,
+                                                        String callBackUrl) {
         String patientId = TokenUtils.readUserId(authorization);
         return validatePatient(patientId)
                 .then(validateConsentRequest(requestId))
                 .flatMap(consentRequest ->
-                        Flux.fromIterable(grantedConsents)
-                                .flatMap(grantedConsent -> {
-                                    var consentArtefact = mapToConsentArtefact(consentRequest, grantedConsent);
-                                    byte[] consentArtefactSignature = getConsentArtefactSignature(consentArtefact);
-                                    return consentArtefactRepository.addConsentArtefactAndUpdateStatus(consentArtefact, requestId, patientId, consentArtefactSignature)
-                                            .thenReturn(Consent.builder().id(consentArtefact.getId()).build());
-                                }).collectList())
-                .map(consents -> ConsentApprovalResponse.builder().consents(consents).build());
+                        generateConsentArtefacts(requestId, grantedConsents, patientId, consentRequest))
+                .flatMap(consents -> {
+                    ConsentApprovalResponse consentApprovalResponse = ConsentApprovalResponse.builder()
+                            .consents(consents)
+                            .build();
+                    return broadcastConsentArtefacts(callBackUrl, consentApprovalResponse.getConsents())
+                            .thenReturn(consentApprovalResponse);
```

Comment:
we could abstract out this part by having a class or something, PostConsentApproval?